portfolio/src/main/java/com/google/sps/data/BookResponseParser.java

```diff
@@ -0,0 +1,169 @@
+package com.google.sps.data;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.data.BookFieldsEnum;
+import com.google.sps.data.Book;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+// Given a json String from Google Book Api, converts the response into a in-memory Book representation.
+class BookResponseParser {
+    private static final String VOLUME_INFO = "volumeInfo";
+    private static final String TITLE = "title";
+    private static final String GENRE = "genre";
+    private static final String CATEGORIES = "categories";
+    private static final String AUTHORS = "authors";
+    private static final String LANGUAGE = "language";
+    private static final String DESCRIPTION = "description";
+    private static final String INFO_LINK = "infoLink";
+    private static final String PAGE_COUNT = "pageCount";
+    private static final String PUBLISHED_DATE = "publishedDate";
+    private static final String PUBLISHER = "publisher";
+    private static final String MATURITY_RATING = "maturityRating";
+    private static final String THUMBNAIL = "thumbnail";
+    private static final String IMAGE_LINKS = "imageLinks";
+    private static final String ISBN = "ISBN_13";
+
+  public Book parseBook(String jsonResponse){
+    JSONObject myResponse = new JSONObject(jsonResponse.toString());
+    return jsonToBook(myResponse);
+  }
+
+  public Book jsonToBook(JSONObject jsonObject) throws JSONException {
+    String tempTitle, tempLang, tempDesc, tempInfoLink, tempPubDate, tempPublisher, tempMatRate;//at same depth
```

Comment:
I'm not sure about this, but idk if you can have multiple variable declarations in one line in the google style guide? 
https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations